File: python/datastructure/otterop/datastructure/string_buffer.py
from otterop.lang.string import String as _String
from otterop.datastructure.linked_list import LinkedList as _LinkedList

class StringBuffer:
    
    def __init__(self):
        self._strings = _LinkedList()

    def add(self, s):
        self._strings.add_last(s)

    def oop_string(self):
        strings = self._strings
        return _String.concat(strings)

File: ts/datastructure/otterop/datastructure/StringBuffer.ts
import { OOPIterable } from '@otterop/lang/OOPIterable';
import { String } from '@otterop/lang/String';
import { LinkedList } from './LinkedList';

export class StringBuffer {

    #strings : LinkedList<String>;

    public constructor() {
        this.#strings = new LinkedList<String>();
    }

    public add(s : String) : void {
        this.#strings.addLast(s);
    }

    public OOPString() : String {
        let strings : OOPIterable<String> = this.#strings;
        return String.concat(strings);
    }
}

